src/main/java/edu/byu/cs/autograder/compile/CompileHelper.java

```diff
@@ -15,26 +15,37 @@ public CompileHelper(GradingContext gradingContext) {
         this.gradingContext = gradingContext;
     }
 
-    private final Collection<StudentCodeInteractor> currentVerifiers =
+    private final Collection<StudentCodeVerifier> currentVerifiers =
             List.of(new ProjectStructureVerifier(), new ModuleIndependenceVerifier());
 
-    private final Collection<StudentCodeInteractor> currentModifiers =
+    private final Collection<StudentCodeModifier> currentModifiers =
             List.of(new PomModifier(), new PassoffJarModifier(), new TestFactoryModifier());
 
     public void compile() throws GradingException {
+        verify();
```

Comment:
Nice work splitting these separate responsibilities into separate functions. I assume that, besides the improvement in overall readability, this also helps the `StudentCodeReader` go out of scope properly?